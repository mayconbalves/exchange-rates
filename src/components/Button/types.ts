export type ButtonProps = {
  disabled?: boolean
  onClick: () => void
  title: string
}
