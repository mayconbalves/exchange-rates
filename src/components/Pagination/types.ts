export type PaginationProps = {
  currentPage: number
  itemsPerPage: number
  totalItems: number
  onPageChange: (page: number) => void
}
