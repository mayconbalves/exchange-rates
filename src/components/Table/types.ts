export type TableDate = {
  rate: string
  value: number
  date: string
  timestamp: number
}

export type SortConfig = {
  key: keyof TableDate | null
  direction: 'ascending' | 'descending'
}

export type TableProps = {
  rates: TableDate[]
  date: string
}
