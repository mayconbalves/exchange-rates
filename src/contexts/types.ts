export type ToastContextProps = {
  showToast: (message: string, type: 'success' | 'error') => void
}
