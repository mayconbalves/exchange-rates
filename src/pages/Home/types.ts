export type Registrations = {
  id: string
  admissionDate: string
  email: string
  employeeName: string
  status: 'REPROVED' | 'APPROVED' | 'REVIEW'
  cpf: string
}
