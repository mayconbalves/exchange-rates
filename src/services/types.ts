export type RateItem = {
  date: string
  rate: string
  value: number
  timestamp: number
}

export type ApiResponse = {
  success: boolean
  timestamp: number
  base: string
  date: string
  rates: Record<string, number>
}
