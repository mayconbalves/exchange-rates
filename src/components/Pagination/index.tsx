import React from 'react'
import Button from '~/components/Button'
import styles from './styles.module.css'
import { PaginationProps } from './types'

const Pagination: React.FC<PaginationProps> = ({
  currentPage,
  itemsPerPage,
  totalItems,
  onPageChange
}) => {
  const totalPages = Math.ceil(totalItems / itemsPerPage)

  const nextPage = () => {
    if (currentPage < totalPages) {
      onPageChange(currentPage + 1)
    }
  }

  const prevPage = () => {
    if (currentPage > 1) {
      onPageChange(currentPage - 1)
    }
  }

  return (
    <div className={styles.wrapper}>
      <Button onClick={prevPage} disabled={currentPage === 1} title="Anterior" />

      <span className={styles.pagination__span}>
        {' '}
        Página {currentPage} de {totalPages}{' '}
      </span>
      <Button onClick={nextPage} disabled={currentPage === totalPages} title="Próxima" />
    </div>
  )
}

export default Pagination
