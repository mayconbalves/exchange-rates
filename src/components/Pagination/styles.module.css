.wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
}

.pagination__button {
  background-color: #ffd862;
  border: 1px solid #000;
  border-radius: 4px;
  cursor: pointer;
  font-size: medium;
  height: 40px;
  margin: 0 10px;
  padding: 5px 10px;
  width: 100px;
}

.pagination__button:disabled {
  border: none;
  background-color: #ccc;
  cursor: not-allowed;
}

.pagination__button:hover:enabled {
  background-color: #fee28e;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.8);
}

.pagination__span {
  font-size: 16px;
}
