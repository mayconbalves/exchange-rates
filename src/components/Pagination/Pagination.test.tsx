import { fireEvent, render, screen } from '@testing-library/react'
import Pagination from '.'
import { PaginationProps } from './types'

describe('Pagination Component', () => {
  const defaultProps: PaginationProps = {
    currentPage: 1,
    itemsPerPage: 5,
    totalItems: 20,
    onPageChange: jest.fn()
  }

  it('Should render pagination component', () => {
    render(<Pagination {...defaultProps} />)
    const paginationElement = screen.getByText(/Página 1 de 4/i)
    expect(paginationElement).toBeInTheDocument()
  })

  it('Should disable "Anterior" button on the first page', () => {
    render(<Pagination {...defaultProps} />)
    const prevButton = screen.getByRole('button', { name: /anterior/i })
    expect(prevButton).toBeDisabled()
  })

  it('Should enable "Próxima" button on the first page', () => {
    render(<Pagination {...defaultProps} />)
    const nextButton = screen.getByRole('button', { name: /próxima/i })
    expect(nextButton).toBeEnabled()
  })

  it('Should disable "Próxima" button on the last page', () => {
    render(<Pagination {...defaultProps} currentPage={4} />)
    const nextButton = screen.getByRole('button', { name: /próxima/i })
    expect(nextButton).toBeDisabled()
  })

  it('Should enable "Anterior" button on pages after the first', () => {
    render(<Pagination {...defaultProps} currentPage={2} />)
    const prevButton = screen.getByRole('button', { name: /anterior/i })
    expect(prevButton).toBeEnabled()
  })

  it('Should call onPageChange with the next page number when "Próxima" button is clicked', () => {
    const onPageChange = jest.fn()
    render(<Pagination {...defaultProps} onPageChange={onPageChange} />)
    const nextButton = screen.getByRole('button', { name: /próxima/i })
    fireEvent.click(nextButton)
    expect(onPageChange).toHaveBeenCalledWith(2)
  })

  it('Should call onPageChange with the previous page number when "Anterior" button is clicked', () => {
    const onPageChange = jest.fn()
    render(<Pagination {...defaultProps} currentPage={2} onPageChange={onPageChange} />)
    const prevButton = screen.getByRole('button', { name: /anterior/i })
    fireEvent.click(prevButton)
    expect(onPageChange).toHaveBeenCalledWith(1)
  })

  it('Should display the correct page number and total pages', () => {
    render(<Pagination {...defaultProps} currentPage={3} />)
    const pageInfo = screen.getByText(/Página 3 de 4/i)
    expect(pageInfo).toBeInTheDocument()
  })

  it('Should be create snapshot', () => {
    const { container } = render(<Pagination {...defaultProps} currentPage={3} />)

    expect(container).toMatchSnapshot()
  })
})
