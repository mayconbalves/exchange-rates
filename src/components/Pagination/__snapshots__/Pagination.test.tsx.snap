// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Should be create snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <button
      class="button"
    >
      Anterior
    </button>
    <span
      class="pagination__span"
    >
       
      Página 
      3
       de 
      4
       
    </span>
    <button
      class="button"
    >
      Próxima
    </button>
  </div>
</div>
`;
